infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -676,25 +682,91 @@ protected void doStartMachine(KubernetesServerResolver serverResolver)
     final Map<String, InternalMachineConfig> machineConfigs = environment.getMachines();
     final String workspaceId = getContext().getIdentity().getWorkspaceId();
     LOG.debug("Begin pods creation for workspace '{}'", workspaceId);
+    PodMerger podMerger = new PodMerger();
+    Map<String, Map<String, Pod>> injectablePods = environment.getInjectablePodsCopy();
     for (Pod toCreate : environment.getPodsCopy().values()) {
       ObjectMeta toCreateMeta = toCreate.getMetadata();
-      final Pod createdPod = namespace.deployments().deploy(toCreate);
+      List<PodData> injectables = getAllInjectablePods(toCreate, injectablePods);
+
+      Pod createdPod;
```

Comment:
Look again :) While the code is indeed very similar, one loop operates on pods and the other on deployments.

The only code that could be extracted from the two loops is the `try` block in the `else` which could have been a method that we would pass the pod data created from either the pod or deployment. Since these are the only 2 occurrences of the logic and they're both in the same method very close together, I didn't consider it necessary to extract a method for the 4 lines of logic.